spec/javascripts/unit/AnalyticsSpec.js

```diff
@@ -213,4 +213,94 @@ describe("GOVUK.Analytics", function () {
+    var $notOnFrameworkString = $('<div class="grid-row" data-analytics=customDimension data-custom-dimension="supplier-not-on-framework">'),
+        $missingDimensionString = $('<div class="grid-row" data-analytics=customDimension>');
+        $canApplyString = $('<h1>Apply for `A cool opportunity`</h1>');
+    
+    afterEach(function () {
+      $notOnFrameworkString.remove();
+      $missingDimensionString.remove();
+      $canApplyString.remove();
+    });
+    
+    it('should send custom dimension if data-custom-dimension', function() {
+      spyOn(GOVUK.GDM.analytics.location, "pathname")
+          .and
+          .callFake(function () {
+            return "/suppliers/opportunities/1/responses/create";
+        });
+
+      $(document.body).append($notOnFrameworkString);
+
+      window.GOVUK.GDM.analytics.pageViews.init();
+
+      expect(window.ga.calls.first().args).toEqual(['set', 'dimension25', 'supplier-not-on-framework']);
+      expect(window.ga.calls.count()).toEqual(2);
+    });
+
+    it('should send custom dimension if eligible to apply', function() {
+      spyOn(GOVUK.GDM.analytics.location, "pathname")
+          .and
+          .callFake(function () {
+            return "/suppliers/opportunities/1/responses/create";
+        });
+
+      $(document.body).append($canApplyString);
+
+      window.GOVUK.GDM.analytics.pageViews.init();
+
+      expect(window.ga.calls.first().args).toEqual(['set', 'dimension25', 'supplier-able-to-apply']);
+      expect(window.ga.calls.count()).toEqual(2);
+    });
+
+    it('should not send custom dimension if no data-custom-dimension or eligible heading', function() {
+      spyOn(GOVUK.GDM.analytics.location, "pathname")
+          .and
+          .callFake(function () {
+            return "/suppliers/opportunities/1/responses/create";
+        });
+
+      window.GOVUK.GDM.analytics.pageViews.setCustomDimensions();
+
+      expect(window.ga.calls.any()).toEqual(false);
+    });
+
+    it('should not send custom dimension if data-analytics tag but data-custom-dimension is missing', function() {
+      spyOn(GOVUK.GDM.analytics.location, "pathname")
+          .and
+          .callFake(function () {
+            return "/suppliers/opportunities/1/responses/create";
+        });
+
+      $(document.body).append($missingDimensionString);
+
+      window.GOVUK.GDM.analytics.pageViews.setCustomDimensions();
+
+      expect(window.ga.calls.any()).toEqual(false);
+    });
+
+    it('should not send custom dimension if not on the create response page', function() {
+      spyOn(GOVUK.GDM.analytics.location, "pathname")
+          .and
+          .callFake(function () {
+            return "/suppliers/opportunities/1/responses";
+        });
+
+      window.GOVUK.GDM.analytics.pageViews.setCustomDimensions();
+
+      expect(window.ga.calls.any()).toEqual(false);
+    });
+
+    it('should only match regex for create response route', function() {
```

Comment:
These are valuable things to test but could we do this by faking calls to `GOVUK.analytics.location.pathname` instead? I think there's some value in not testing the internals of our analytics code if possible (mainly informed by the talk we saw recently on this - https://www.youtube.com/watch?v=URSWYvyc42M).